programs/bpf_loader/src/syscalls.rs

```diff
@@ -2540,6 +2574,94 @@ fn call<'a>(
     Ok(SUCCESS)
 }
 
+// Return data hanlding
+pub struct SyscallSetReturnData<'a> {
+    invoke_context: Rc<RefCell<&'a mut dyn InvokeContext>>,
+    loader_id: &'a Pubkey,
+}
+impl<'a> SyscallObject<BpfError> for SyscallSetReturnData<'a> {
+    fn call(
+        &mut self,
+        addr: u64,
+        len: u64,
+        _arg3: u64,
+        _arg4: u64,
+        _arg5: u64,
+        memory_mapping: &MemoryMapping,
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        if len > MAX_RETURN_DATA {
```

Comment:
Consume first before doing anything else on the assumption that things will work out

Reply:
Changed